Sources/TUSKit/Files.swift

```diff
@@ -153,11 +161,17 @@ final class Files {
     /// - Throws: Any error from FileManager when removing a file.
     func removeFileAndMetadata(_ metaData: UploadMetadata) throws {
         let filePath = metaData.filePath
-        let metaDataPath = metaData.filePath.appendingPathExtension("plist")
+        let fileName = filePath.lastPathComponent
+        let metaDataPath = storageDirectory.appendingPathComponent(fileName).appendingPathExtension("plist")
         
         try queue.sync {
-            try FileManager.default.removeItem(at: filePath)
             try FileManager.default.removeItem(at: metaDataPath)
+#if os(macOS)
```

Comment:
Do we actually need this check? It looks like for non-sandboxed macOS apps this would cause the temporary file to not be deleted while I do believe that’s what we should be doing. It’s just where the file that we need to delete is located that’s different, right?

Reply:
In fact, I recommend not copying and uploading files on macOS, and I will update pr when I have the new code ready

Reply:
Sounds good, I will keep the PR open until then

Reply:
I removed the sandbox environment and blocked the copying of temporary files on macOS to avoid running out of disk space for large file uploads.